certora/specs/RevertsMintBurnEthereum.spec

```
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+methods {
+    function owner() external returns address envfree;
+    function totalSupply() external returns uint256 envfree;
+    function balanceOf(address) external returns uint256 envfree;
+    function delegatee(address) external returns address envfree;
+    function delegatedVotingPower(address) external returns uint256 envfree;
+}
+
+// Check the revert conditions for the burn function.
+rule mintRevertConditions(env e, address to, uint256 amount) {
+    mathint totalSupplyBefore = totalSupply();
+    uint256 balanceOfSenderBefore = balanceOf(e.msg.sender);
+    uint256 toVotingPowerBefore = delegatedVotingPower(delegatee(to));
+
+    // Safe require as zero address can't possibly delegate voting power.
+    require delegatee(0) == 0;
+
+    // Safe require because if the delegatee is not zero the recipient's delegatee's voting power is lesser than or equal to the total supply of tokens
+    require delegatee(to) != 0 => toVotingPowerBefore <= totalSupply();
+
+    mint@withrevert(e, to, amount);
+    assert lastReverted <=> e.msg.sender != owner() || to == 0 || e.msg.value != 0 || totalSupplyBefore + amount > max_uint256;
+}
+
+// Check the revert conditions for the burn function.
+rule burnRevertConditions(env e, uint256 amount) {
+    uint256 balanceOfSenderBefore = balanceOf(e.msg.sender);
+    uint256 senderVotingPowerBefore = delegatedVotingPower(delegatee(e.msg.sender));
```

it's not the voting power of the sender right?
```suggestion
    uint256 delegateeVotingPowerBefore = delegatedVotingPower(delegatee(e.msg.sender));
```

---

Indeed, fixed in https://github.com/morpho-org/morpho-token-upgradeable/pull/98/commits/1b679307ba4245b7d591ba01045c16ca01ef07f4.